main/api/dependencies/auth.py

```diff
@@ -34,8 +35,12 @@ async def require_authenticated_user(
     return user
 
 
-async def require_permission_on_category(
-    user: UserModel = Depends(require_authenticated_user), category: CategoryModel = Depends(get_category_by_id)
-) -> None:
-    if category.user_id != user.id:
-        raise ForbiddenException("User does not have permission to perform this action")
+def require_ownership(dependency) -> Callable:
+    def inner_work(
```

Comment:
This is a bad naming. We should name the function in a way that we can understand the purpose of function just by reading the function name, and that is called clean code